sdk/identity/Azure.Identity/src/ChainedTokenCredential.cs

```diff
@@ -22,6 +22,14 @@ public class ChainedTokenCredential : TokenCredential
 
         private readonly TokenCredential[] _sources;
 
+        /// <summary>
+        /// Constructor for instrumenting in tests
+        /// </summary>
+        internal ChainedTokenCredential()
```

Comment:
This should probably be `protected internal` instead of just `internal` for mocking purposes

Reply:
We have another public constructor, won't that be enough?

Reply:
The public constructor requires you to pass one or more credential instances or it will throw. It's not a big deal since they could pass arbitrary instances and ignore them in their implementation. But it would just be more convenient if this were protected. Also, it adheres to the guideline, https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking-constructor.

Reply:
This constructor doesn't initialize `ChainedTokenCredential` properly, making it unusable. We either have to make it fully functional with `_sources == null`, which IMO is meaningless, or leave it internal for testing purposes.

Reply:
As this would be for mocking purposes I wouldn't expect developers to use it unless they were overriding `GetToken` and `GetTokenAsync` as well. It would need to be documented as such as well.